R/lbt05.R

```diff
@@ -33,12 +38,24 @@ lbt05_main <- function(adam_db,
   lbl_anrind <- var_labels_for(adam_db$adlb, "ABN_DIR")
   lbl_param <- var_labels_for(adam_db$adlb, "PARAM")
 
-  map <- expand.grid(
-    PARAM = levels(adam_db$adlb$PARAM),
-    ABN_DIR = c("Low", "High"),
-    stringsAsFactors = FALSE
-  ) %>%
-    arrange(.data$PARAM, desc(.data$ABN_DIR))
+  map <- if (is.null(map)) {
+    expand.grid(
+      PARAM = levels(adam_db$adlb$PARAM),
+      ABN_DIR = c("Low", "High"),
+      stringsAsFactors = FALSE
+    ) %>%
+      arrange(.data$PARAM, desc(.data$ABN_DIR))
```

Comment:
i was wondering if we make this as a function? as the default mapping?

so the following line for else condition, they are executed regardless?

Reply:
what if we do the following then

```
if (is.null(map)){
  map <- expand.grid(
      PARAM = levels(adam_db$adlb$PARAM),
      ABN_DIR = c("Low", "High"),
      stringsAsFactors = FALSE
    ) 
}

matching <- adam_db$adlb %>%
      dplyr::select(.data$PARAMCD, .data$PARAM) %>%
      dplyr::distinct()

    map %>%
      dplyr::filter(.data$PARAMCD %in% matching$PARAMCD) %>%
      dplyr::left_join(matching, by = "PARAMCD") %>%
      dplyr::select(.data$PARAM, .data$ABN_DIR) %>%
      mutate(across(everything(), as.character)) %>%
      arrange(.data$PARAM, desc(.data$ABN_DIR)
```